pkg/solana/chainwriter/helpers.go

```diff
@@ -0,0 +1,198 @@
+package chainwriter
+
+import (
+	"context"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/gagliardetto/solana-go"
+	"github.com/gagliardetto/solana-go/rpc"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/utils"
+)
+
+type TestArgs struct {
+	Inner []InnerArgs
+}
+
+type InnerArgs struct {
+	Address []byte
+}
+
+type DataAccount struct {
+	Discriminator        [8]byte
+	Version              uint8
+	Administrator        solana.PublicKey
+	PendingAdministrator solana.PublicKey
+	LookupTable          solana.PublicKey
+}
+
+// GetValuesAtLocation parses through nested types and arrays to find all locations of values
+func GetValuesAtLocation(args any, location string) ([][]byte, error) {
```

Comment:
Callers in multiple locations are trying to access response slice elem by index without checking the size which is panic prone

Reply:
I believe the only panic prone one was the debugID check, which I've fixed now... I think all the ones in `traversePath` are safe. Let me know if you think otherwise